compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -832,11 +834,12 @@ object Contexts {
     store = initialStore
       .updated(settingsStateLoc, settingsGroup.defaultState)
       .updated(notNullInfosLoc, Nil)
+      .updated(compilationUnitLoc, NoCompilationUnit)
     searchHistory = new SearchRoot
     gadt = EmptyGadtConstraint
   }
 
-  @sharable object NoContext extends Context(null) {
+  @sharable object NoContext extends Context(null.asInstanceOf) {
```

Comment:
Currently, we couldn't use uncheckedNN here, because type augument `T` is infered to `Nothing`. I think we need to improve the inference to handle this case.

Reply:
Maybe this would work instead?
```scala
(null: Context | Null).uncheckedNN
```

Reply:
Done